### PR TITLE
chore: logging 3 - remove prints from tools directory

### PR DESCRIPTION
**Pull request by hollandjg**

### CODE REVIEW

File: src/icesat2_tracks/ICEsat2_SI_tools/angle_optimizer.py [hunk 1]
```diff
@@ -232,8 +236,8 @@ def optimize(self, fitting_args= None , method='dual_annealing', verbose= True):
         self.fitter_optimize = self.LM.minimize(self.objective_func, self.params,  method=method,
                         args=fitting_args, kws=fitting_kargs )
         if verbose:
-            print(self.LM.report_fit(self.fitter_optimize))
-            print('results at self.fitter_optimize')
+            _logger.debug("%s", self.LM.report_fit(self.fitter_optimize))
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/angle_optimizer.py:239]:
```suggestion
            _logger.debug("fit report: %s", self.LM.report_fit(self.fitter_optimize))
```

File: src/icesat2_tracks/ICEsat2_SI_tools/filter_regrid.py [hunk 2]
```diff
@@ -448,7 +450,7 @@ def bin_means(T2, dist_grid):
 
     for i in np.arange(1,ilim-1, 1):
         if i % 5000 ==0:
-            print(i)
+            _logger.debug(i)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/filter_regrid.py:453]:
```suggestion
            _logger.debug("i= %s", i)
```

File: src/icesat2_tracks/ICEsat2_SI_tools/generalized_FT.py [hunk 3]
```diff
@@ -397,18 +398,18 @@ def calc_gFT_apply(stancil, prior):
...
                 prior = I_return["PSD"], I_return["weight"]
             # 2nd step
             if prior[0] is False:
-                print("1st GFD failed (priors[0]=false), skip 2nd step")
+                _logger.debug("1st GFD failed (priors[0]=false), skip 2nd step")
             else:
-                print("2nd step: use set priors:", type(prior[0]), type(prior[1]))
-                print(prior[0][0:3], prior[1][0:3])
+                _logger.debug("2nd step: use set priors: %s %s", type(prior[0]), type(prior[1]))
+                _logger.debug("%s %s", prior[0][0:3], prior[1][0:3])
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/generalized_FT.py:412]:
```suggestion
                _logger.debug("2nd step: use set priors of types: %s %s", type(prior[0]), type(prior[1]))
                _logger.debug("first three elements of priors: %s %s", prior[0][0:3], prior[1][0:3])
```

File: src/icesat2_tracks/ICEsat2_SI_tools/iotools.py [hunk 4]
```diff
@@ -75,7 +79,7 @@ def init_data(ID_name, batch_key, ID_flag, ID_root, prefix="A01b_ID"):
 
     """
 
-    print(ID_name, batch_key, ID_flag)
+    _logger.debug("%s %s %s",ID_name, batch_key, ID_flag)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/iotools.py:82]:
```suggestion
    _logger.debug("id name: %s, batch key: %s, id flag: %s", ID_name, batch_key, ID_flag)
```

File: src/icesat2_tracks/ICEsat2_SI_tools/iotools.py [hunk 5]
```diff
@@ -298,10 +302,10 @@ def nsidc_icesat2_get_associated_file(
         colnames, collastmod, colerror = icesat2_toolkit.utilities.nsidc_list(
             PATH, build=False, timeout=TIMEOUT, parser=parser, pattern=R1, sort=True
         )
-        print(colnames)
+        _logger.debug(colnames)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/iotools.py:305]:
```suggestion
        _logger.debug("colnames: %s", colnames)
```

File: src/icesat2_tracks/ICEsat2_SI_tools/iotools.py [hunk 6]
```diff
@@ -298,10 +302,10 @@ def nsidc_icesat2_get_associated_file(
...
         # -- print if file was not found
         if not colnames:
-            print(colerror)
+            _logger.debug(colerror)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/iotools.py:308]:
```suggestion
            _logger.debug("colerror: %s", colerror)
```

File: src/icesat2_tracks/ICEsat2_SI_tools/spectral_estimates.py [hunk 7]
```diff
@@ -1021,11 +1026,11 @@ def __init__(self,data, x, L, ov=None, window=None, save_chunks=False, plot_chun
...
-                    print('end of TS is reached')
-                    print('last spec No: '+str(last_k))
-                    print('spec container: '+str(specs.shape))
-                    print('last used Timestep: '+str(last_used_TS))
-                    print('length of TS '+ str(dsize) +'ms')
+                    _logger.debug('end of TS is reached')
+                    _logger.debug('last spec No: %s', str(last_k))
+                    _logger.debug('spec container: %s', str(specs.shape))
+                    _logger.debug('last used Timestep: %s', str(last_used_TS))
+                    _logger.debug('length of TS %s ms', str(dsize))
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/spectral_estimates.py:1033]:
```suggestion
                    _logger.debug('end of TS is reached')
                    _logger.debug('last spec No: %s', last_k)
                    _logger.debug('spec container: %s', specs.shape)
                    _logger.debug('last used Timestep: %s', last_used_TS)
                    _logger.debug('length of TS %s ms', dsize)
```

File: src/icesat2_tracks/ICEsat2_SI_tools/spicke_remover.py [hunk 8]
```diff
@@ -21,18 +25,22 @@ def spicke_remover(data, nstd=20.0, spreed=500.0, max_loops=10.0 , verbose=False
         if  nstd* datastd < np.nanmax(np.abs(data2)):
             act_flag=True
             if verbose:
-                print('true: '+ str(nstd* datastd) +' < '+str( np.nanmax(np.abs(data)) ) )
+                _logger.debug('true: %s < %s',
+                              str(nstd* datastd),
+                              str( np.nanmax(np.abs(data))))
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/spicke_remover.py:30]:
```suggestion
                _logger.debug('true: %s < %s',
                              nstd* datastd,
                              np.nanmax(np.abs(data)))
```

File: src/icesat2_tracks/ICEsat2_SI_tools/spicke_remover.py [hunk 9]
```diff
@@ -21,18 +25,22 @@ def spicke_remover(data, nstd=20.0, spreed=500.0, max_loops=10.0 , verbose=False
...
             data2=spickes_to_mean(data2, nloop=0, spreed=spreed, gaussian=False)
             looper_count+=1
         else:
             if verbose:
-                print('False: '+ str(nstd* datastd) +' > '+str( np.nanmax(np.abs(data)) ) )
+                _logger.debug('False: %s > %s ',
+                              str(nstd* datastd),
+                              str( np.nanmax(np.abs(data))))
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/spicke_remover.py:37]:
```suggestion
                _logger.debug('False: %s > %s ',
                              nstd* datastd,
                              np.nanmax(np.abs(data)))
```

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
Some suggested cleanups here.

File: src/icesat2_tracks/ICEsat2_SI_tools/spectral_estimates.py [hunk 10]
```diff
@@ -762,11 +767,11 @@ def calc_spectrum_and_field_apply(stancil):
         # apply func to all stancils
         # Spec_returns=list()
         # for ss in stancil_iter:
-        #     print(ss)
+        #     _logger.debug(ss)
         #     Spec_returns.append( calc_spectrum_apply(ss) )
 
         map_func = map if map_func is None else map_func
-        print(map_func)
+        _logger.debug(map_func)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/ICEsat2_SI_tools/spectral_estimates.py:774]: I am not sure this one is useful.
```suggestion
```

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
One more change here.

**Review by cpaniaguam — DISMISSED** [review]
All good!

**Review by kmilo9999 — DISMISSED** [review]

File: src/icesat2waves/tools/iotools.py [hunk 11]
```diff
@@ -328,7 +332,7 @@ def ATL03_download(username, password, dpath, product_directory, sd, file_name):
     """
 
     HOST = ["https://n5eil01u.ecs.nsidc.org", "ATLAS", product_directory, sd, file_name]
-    print("download to:", dpath + "/" + HOST[-1])
+    _logger.debug("download to: %s", dpath + "/" + HOST[-1])
```
[the remark this answers is not in the document]
Comment (hollandjg) [reply src/icesat2waves/tools/iotools.py:335]: Wow, that works? It looks like a syntax error to me. I didn't know you could concatenate strings that way. How does Black reformat it? I'm not super keen – I don't think the suggestion makes the syntax any clearer.

[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2waves/tools/iotools.py:335]: Maybe it doesn't work. I guess it only works with literals.

**Pull request merged**